Test/AwsCommonRuntimeKitTests/http/HTTP2ClientConnectionTests.swift

```diff
@@ -0,0 +1,184 @@
+////  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+////  SPDX-License-Identifier: Apache-2.0.
+import XCTest
+@testable import AwsCommonRuntimeKit
+
+class HTTP2ClientConnectionTests: HTTPClientTestFixture {
+
+    let expectedVersion = HTTPVersion.version_2
+
+    func testGetHTTP2RequestVersion() async throws {
+        let connectionManager = try await getHttpConnectionManager(endpoint: "httpbin.org", alpnList: ["h2","http/1.1"])
+        let connection = try await connectionManager.acquireConnection()
+        XCTAssertEqual(connection.httpVersion, HTTPVersion.version_2)
+    }
+
+    // Test that the binding works not the actual functionality. C part has tests for functionality
+    func testHTTP2UpdateSetting() async throws {
+        let connectionManager = try await getHttpConnectionManager(endpoint: "httpbin.org", alpnList: ["h2","http/1.1"])
+        let connection = try await connectionManager.acquireConnection()
+        if let connection = connection as? HTTP2ClientConnection {
+            try await connection.updateSetting(setting: HTTP2Settings(enablePush: false))
```

Comment:
there should be an assertion after `updateSetting` call

in rest of the tests too.

Reply:
I am not sure what to assert. These tests just check that the bindings work and the C has tests for actual functionality. @TingDaoK Is there something we can assert for in these tests?

Reply:
In that case your would want to use https://developer.apple.com/documentation/xctest/xctassertnothrow if no error is thrown in updateSetting

Reply:
It doesn't work with error `'async' call in an autoclosure that does not support concurrency`. The test will fail if anything throws anyhow.